src/types.mligo

```diff
@@ -161,6 +172,8 @@ type storage =
   ; freeze_history : freeze_history
```

Comment:
While writing the previous comments I realized that we don't need to store `frozen_supply` inside of `quorum_threshold_at_stage` at all (I didn't edit the previous comments as they maybe show the point better).

We already know that value right here, in `total_supply`.

If we remember to update the `quorum_threshold_at_stage` every time we `freeze` or `unfreeze`, we already know that in period `x` the `total_supply` of `frozen_token_id` here in the storage is the same as either:
1. the total_supply of frozen tokens at period `x - 1` (that we can use to update the quorum)
2. the total_supply of frozen tokens at period `x` (that we don't need to use as the quorum is already up-to-date)

Reply:
@pasqu4le I doubt.

>>We already know that value right here, in total_supply.

To be clear how I understand your sugeestion, is that in `freeze` entrypoint, we can possibly look at the `quorum_threshold_at_stage`, and when we see that its `updated_at` period is less than current one, we can use the current `total_supply` as the number of frozen_tokens in last period, and use it to update the quorum_threshold, and then proceed to actually freeze the tokens.

But doesn't `total_supply` at this point  include tokens that were frozen in all period upto the current one, and not limited to tokens that were frozen in (x -1)? 

Or doesn't that matter at all?